### PR TITLE
propagate errors from Type.size()

### PR DESCRIPTION
**Pull request by WalterBright**
To rely less on the global error count.

### CODE REVIEW

File: src/dmd/dcast.d [hunk 1]
```diff
@@ -1728,6 +1728,11 @@ Expression castTo(Expression e, Scope* sc, Type t, Type att = null)
                     }
                     const fsize = t1b.nextOf().size();
                     const tsize = tob.nextOf().size();
+                    if (fsize == SIZE_INVALID || tsize == SIZE_INVALID)
+                    {
+                        result = ErrorExp.get();
```
Comment (ljmf00) [inline src/dmd/dcast.d:1733]: When do `SIZE_INVALID` happen? Can we add tests to cover these branches?

Comment (WalterBright) [reply src/dmd/dcast.d:1733]: Generally with unresolvable forward references. They're kinda hard to trigger, as a lot of work has gone into figuring out ways to resolve forward references.

Comment (ibuclaw) [reply src/dmd/dcast.d:1733]: Indeed, if you find a test, that should rather go in a bug report so that it it can be looked into and fixed.

**Comment by UplinkCoder** [comment]
relaying less on the global errors is a very useful road to go down indeed.

**Review by ibuclaw — APPROVED** [review]

**Pull request merged**